File: ProvaVue/src/interfaces/heroi.ts
export interface Superpoder {
  id: number;
  superpoder: string;
  descricao: string;
}

export interface Heroi {
  id: number;
  nome: string;
  nomeHeroi: string;
  dataNascimento: string;
  altura: number;
  peso: number;
  superpoderes: Superpoder[];
}

export interface FormHeroi {
  nome: string;
  nomeHeroi: string;
  dataNascimento: string;
  altura: number | null | string;
  peso: number | null | string;
  superpoderesIds: number[];
}

File: WebApplication/Models/Heroi.cs
using System.ComponentModel.DataAnnotations;

namespace ViceriSeidorHero.Models
{
    public class Heroi
    {
        public int Id { get; set; }

        [Required, MaxLength(120)]
        public string Nome { get; set; } = null!;

        [Required, MaxLength(120)]
        public string NomeHeroi { get; set; } = null!;

        [Required]
        public DateTime DataNascimento { get; set; }

        [Required]
        public double Altura { get; set; }

        [Required]
        public double Peso { get; set; }

        public ICollection<HeroiSuperpoder> HeroisSuperpoderes { get; set; } = new List<HeroiSuperpoder>();
    }
}
